lib/utils.js

```diff
@@ -178,31 +178,43 @@ function endOfMonthTimestamp() {
   return Date.UTC(curDate.getUTCFullYear(), curDate.getUTCMonth() + 1, 1, 0, 0, 0, 0);
 }
 
-function resolveElevatedParams(erlParams, bucketKeyConfig) {
+function resolveElevatedParams(erlParams, bucketKeyConfig, key) {
   // provide default values for elevated_limits unless the bucketKeyConfig has them
-  return {
+  const elevatedLimits = {
     ms_per_interval: bucketKeyConfig.ms_per_interval,
     size: bucketKeyConfig.size,
     erl_activation_period_seconds: 0,
     erl_quota: 0,
     erl_quota_interval: 'quota_per_calendar_month',
-    erl_is_active_key: 'defaultActiveKey',
-    erl_quota_key: 'defaultQuotaKey',
+    erl_is_active_key: 'ERLActiveKey',
+    erl_quota_key: 'ERLQuotaKey',
     ...erlParams,
     ...bucketKeyConfig.elevated_limits,
     erl_configured_for_bucket: !!(erlParams && bucketKeyConfig.elevated_limits?.erl_configured_for_bucket),
   };
+
+  elevatedLimits.erl_is_active_key = replicateHashtag(key, elevatedLimits.erl_is_active_key)
+  elevatedLimits.erl_quota_key = replicateHashtag(key, elevatedLimits.erl_quota_key)
+
+  return elevatedLimits;
 }
 
-class LimitdRedisConfigurationError extends Error {
-  constructor(msg, extra) {
-    super();
-    this.name = this.constructor.name;
-    this.message = msg;
-    Error.captureStackTrace(this, this.constructor);
-    if (extra) {
-      this.extra = extra;
-    }
+function replicateHashtag(baseKey, key) {
+  const idxOpenBrace = baseKey.indexOf('{')
+  if (idxOpenBrace < 0) {
+    return key + `:{${baseKey}}`;
+  }
+
+  const idxCloseBrace = baseKey.indexOf('}', idxOpenBrace)
```

Comment:
have you checked behaviour of something like `a}{b`?

Just because using `indexOf(searchString, offsetIndex)` is prone to off-by-one type errors - I can't remember if the offset is inclusive or exclusive. Would be good to make sure we have a test case.

Reply:
we have [this](https://github.com/auth0/limitd-redis/blob/3282eef87c6a20df73c1bdc4b6430b83d8e17786/test/utils.tests.js#L345-L349) test, but added a shorter version just in case